ui-v2/app/templates/dc/services/show/upstreams.hbs

```diff
@@ -8,9 +8,17 @@
     {{#let item.Service.Namespace as |nspace|}}
       <ListCollection @items={{gatewayServices}} class="consul-upstream-list" as |item index|>
       {{#if (service/exists item)}}
-        <a data-test-service-name href={{href-to 'dc.services.show' item.Name}}>
-          {{item.Name}}
-        </a>
+        {{#if (env 'CONSUL_NSPACES_ENABLED')}}
+          {{#if (not-eq item.Namespace nspace)}}
+            <a data-test-service-name href="{{concat '/ui/~' item.Namespace '/' dc '/services/' item.Name}}">
+              {{item.Name}}
```

Comment:
We need to stick to using `href-to` here. I think you can use the route `nspace.dc.services.show` if you want to link to a specific namespace (if you don't use the `nspace.` prefix it will use your current namespace). I would imagine this approach is done elsewhere. If you get stuck with it gimme a yell later.